safenode/src/node/api.rs

```diff
@@ -113,118 +112,123 @@ impl Node {
             }
-            // This should be unreachable, as we only return protocol errors.
-            other => other?,
-        };
-
-        let response = match self
-            .storage
-            .write(&Cmd::Dbc {
+            Cmd::Register(cmd) => {
+                let result = self.registers.write(&cmd).await;
+                match cmd {
+                    RegisterCmd::Create(_) => CmdResponse::CreateRegister(result),
+                    RegisterCmd::Edit(_) => CmdResponse::EditRegister(result),
+                }
+            }
+            Cmd::SpendDbc {
                 signed_spend,
                 source_tx,
-            })
-            .await
-        {
-            CmdResponse::Spend(Err(ProtocolError::DoubleSpendAttempt { new, existing })) => {
-                warn!("Double spend attempted! New: {new:?}. Existing:  {existing:?}");
+                fee_ciphers,
+            } => {
+                // First we fetch all parent spends from the network.
+                // They shall naturally all exist as valid spends for this current
+                // spend attempt to be valid.
+                let parent_spends = match self.get_parent_spends(source_tx.as_ref()).await {
+                    Ok(parent_spends) => parent_spends,
+                    Err(Error::Protocol(err)) => return CmdResponse::Spend(Err(err)),
+                    Err(error) => {
+                        return CmdResponse::Spend(Err(ProtocolError::Transfers(
+                            TransferError::SpendParentCloseGroupIssue(error.to_string()),
+                        )))
+                    }
+                };
 
-                let request =
-                    Request::Event(Event::double_spend_attempt(new.clone(), existing.clone())?);
-                let _resp = self.send_to_closest(&request).await?;
+                // Then we try to add the spend to the transfers.
+                // This will validate all the necessary components of the spend.
+                let res = match self
+                    .transfers
+                    .try_add(signed_spend, source_tx, fee_ciphers, parent_spends)
+                    .await
+                {
+                    Err(TransferError::DoubleSpendAttempt { new, existing }) => {
+                        warn!("Double spend attempted! New: {new:?}. Existing:  {existing:?}");
+                        if let Ok(event) =
+                            Event::double_spend_attempt(new.clone(), existing.clone())
+                        {
+                            match self.send_to_closest(&Request::Event(event)).await {
+                                Ok(_) => {}
+                                Err(err) => {
+                                    warn!("Failed to send double spend event to closest peers: {err:?}");
+                                }
+                            }
+                        }
+
+                        Err(ProtocolError::Transfers(
```

Comment:
Not sure what you are asking. But this error is very important, because it's what will have us tell all other close nodes that there was a double spend attempt, and they will update accordingly (after verifying it).

We can't really do much if we can't send it to them. We could add a retry. And a retry of the retry. And a retry on that.... There's no end to it.

The simple approach: If we can't send it now, we will have to rely on them seeing the exact same thing when they also got this request. Because, that's actually what will most likely happen in 99.9999...% of cases.
This notification to them is like a backup thing, to be _really_ sure.